### DISCUSSION TITLE
Changing Filepath of Image textures

### DISCUSSION THREAD

**Discussion by MadeOfClicks** [Gorgious56/asset_browser_utilities#86] [Ideas]
Hey,
So what bothers me with the asset browser is that PBR materials have different resolutions and I dont want to clutter the asset browser with 3 to 5 versions of the same Material (1/2k, 1k, 2k, ..)
Making different Material Libraries for each Resolution is also not ideal in my opinion

I'm currently investigating how epbr and materialiq handle switching the resolutions of their materials and am at the point of trying to learn python to get this to work somehow, I mean implementing buttons to search and replace e.g. 2k for 4k but I'm not sure I'm getting there as a noob

Do you think this is something you want to have a look into?
I've not found an addon providing this feature so your kind of my last hope before getting into it myself
Best Regards

**Comment by Gorgious56** [comment]
Hello ! I think this is a bit out of scope for this add-on but I'll be glad to help you out. So you have a bunch of materials using a set of textures and you'd like to swap them at runtime from a set of higher resolution texture ?

**Reply by MadeOfClicks** [reply]
Yes, 
so basically that's what I want but both ways, so swapping images out for higher/lower res.
I want to have buttons in the material properties to do so with one click.

At least that's how I imagine it to be most convenient.

For context: I've renamed and recategorized all materials I've had, from add-ons and downloaded.
I've only used the highest resolution textures of each for the asset browser, but well that's overkill and I've got to swap them out for the lower resolution ones most of the time, which is a pain, even with Node Wrangler.

Epbr has implemented it in it's addon for their filestructure, whitch seperates the textures into folders for each Resolution.
Pretty much the same as when you download textures from any site.

Materialiq on the other hand has all it's textures in one folder and swaps them out by filtering depending on their naming convention..

I've already got Serpens and easybpi and I guess both mentioned addon creators wouldn't mind if I copy parts of their solutions to tinker with it, if I contact them but yea, I will have  to deep dive into it and we'll see, I'll gladly take your offer and might ask a question when I hit a roadblock

**Comment by Gorgious56** [comment]
Yeah I think that's a bit too complicated for this addon unfortunately. :) But don't hesitate to ping me here for information or over Blender Stack Exchange where I and several other talented people answer questions regularly, that way it can benefit everyone in the long term, too.

Cheers
